gridcapa-cse-valid-app/src/main/java/com/farao_community/farao/cse_valid/app/CseValidNetworkShifter.java

```diff
@@ -0,0 +1,103 @@
+                                                         Network network,
+                                                         String glskUrl) {
+        return getNetworkShifter(getSplittingFactorsForFullImport(timestampWrapper), network, glskUrl);
+    }
+
+    public NetworkShifter getNetworkShifterForExportCornerWithItalyFrance(TTimestampWrapper timestampWrapper,
+                                                                          Network network,
+                                                                          String glskUrl) {
+        return getNetworkShifter(getSplittingFactorsForExportCornerWithItalyFrance(timestampWrapper), network, glskUrl);
+    }
+
+    NetworkShifter getNetworkShifterForExportCornerWithAllCountries(TTimestampWrapper timestampWrapper,
+                                                                    Network network,
+                                                                    String glskUrl) {
+        return getNetworkShifter(getSplittingFactorsForExportCornerWithAllCountries(timestampWrapper), network, glskUrl);
+    }
+
+    Map<String, Double> getSplittingFactorsForFullImport(TTimestampWrapper timestampWrapper) {
+        Map<String, Double> splittingFactorsMap = timestampWrapper.getImportCornerSplittingFactors();
+        splittingFactorsMap.put(eicCodesConfiguration.getItaly(), -1.);
+        return splittingFactorsMap;
+    }
+
+    Map<String, Double> getSplittingFactorsForExportCornerWithItalyFrance(TTimestampWrapper timestampWrapper) {
+        Map<String, Double> result = new HashMap<>();
+        double franceFactor = timestampWrapper.isFranceImportingFromItaly() ? -1.0 : 1.0;
+        result.put(eicCodesConfiguration.getFrance(), franceFactor);
+        result.put(eicCodesConfiguration.getItaly(), franceFactor * -1);
+        return result;
+    }
+
+    Map<String, Double> getSplittingFactorsForExportCornerWithAllCountries(TTimestampWrapper timestampWrapper) {
+        return timestampWrapper.getExportCornerSplittingFactors().entrySet().stream()
+                .collect(Collectors.toMap(
+                    Map.Entry::getKey,
+                    stringDoubleEntry -> stringDoubleEntry.getValue() * getFactorSignOfCountry(timestampWrapper, stringDoubleEntry.getKey())
+                ));
+    }
+
+    private double getFactorSignOfCountry(TTimestampWrapper timestampWrapper, String countryEic) {
+        return timestampWrapper.isCountryImportingFromItaly(countryEic) ? -1 : 1;
+    }
+
+    public void shiftNetwork(double shiftValue, Network network, TTimestampWrapper timestampWrapper, String glskUrl) {
+        NetworkShifter networkShifter = getNetworkShifterForExportCornerWithAllCountries(timestampWrapper, network, glskUrl);
+        try {
+            networkShifter.shiftNetwork(shiftValue, network);
+        } catch (GlskLimitationException | ShiftingException e) {
+            throw new CseValidShiftFailureException(e.getMessage());
```

Comment:
Add a Throwable to the constructor so you can propagate the initial exeception especially to have the stack, otherwise we will lose information